packages/firestore/src/model/mutation.ts

```diff
@@ -188,6 +202,15 @@ export class Precondition {
  * DeleteMutation     NoDocument(v3)        NoDocument(v0)
  * DeleteMutation     null                  NoDocument(v0)
  *
+ * For acknowledged mutations, we use the updateTime of the WriteResponse as
+ * the resulting version for Set, Patch, and Transform mutations. As deletes
+ * have no explicit update time, we use the commitTime of the WriteResponse for
+ * Delete mutations.
+ *
+ * If due to precondition mismatches an acknowledged mutation can't be applied
```

Comment:
This language is hard to understand. How about:

If a mutation is acknowledged but locally fails its precondition, we return an `UnknownDocument` and rely on Watch to send us the updated version.

Reply:
Simplified to:

```
 * If a mutation is acknowledged by the backend but fails the precondition check
 * locally, we return an `UnknownDocument` and rely on Watch to send us the
 * updated version.
```